src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -1009,6 +1009,50 @@ void testCheckOutItemHoldItem_invalidTransactionState() {
     assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
   }
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql"
+  })
+  void returnTransactionByBarcodeAndState_when_transactionsFound() {
+    var responseEntity = testRestTemplate.getForEntity(
+        "/inn-reach/transactions/search?shippedItemBarcode={shippedItemBarcode}&transactionStates={transactionStates}",
+        InnReachTransactionsDTO.class, "ABC-abc-1234", new String[] {"PATRON_HOLD", "ITEM_HOLD"});
+
+    assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+
+    var body = responseEntity.getBody();
+
+    assertNotNull(body);
+
+    var transactions = body.getTransactions();
+
+    assertNotNull(transactions);
+    assertFalse(transactions.isEmpty());
```

Comment:
I think it would also make sense to check that transactions that were found match searching parameters and also are sorted in right order.